robot_sort/robot_sort.py

```diff
@@ -129,46 +128,14 @@ def sort(self):
                     self.move_left()
                     self.swap_item()
                     self.move_right()
-                        
-            while self.can_move_left():
-                if self.compare_item() == None:
-                    self.swap_item()
-                    self.move_left()
-                    
-                if self.compare_item() == 1:
-                    self.move_right()
-                    self.swap_item()
-                    self.move_left()
+            # after the robot reaches the end of the array the main while loop will continue from here 
                     
-                if self.compare_item() == 0:  #if the value of item the robot holding is greater than the value of the item on the list that the robot is comparing
-                    self.move_left()
-                    self.swap_item()
-                    self.move_left()
-                
-                if self.compare_item() == -1: #if the robot is holding has less than value
-                    self.swap_item()
-                    self.move_right()
-                    self.swap_item()
-                    self.move_left()
-                    self.set_light_on()
+            if self.light_is_on() and self.can_move_right() is False: #break
+                while self.can_move_left() == True:
+                    self.move_left() #will be activated
         print('Robot works')
-                    
-        
-            # while self.move_right:
-            #     if self.compare_item == 1: 
-            #         self.swap_item()
-            #         self.move_left
-            #     else: 
-            #         self.move_right()
-            #         #when do I switch off the light? LOLZ
-        
-            # while self.can_move_right():
-            #     if self.compare_item == -1:
-            #         self.swap_item()
-            #         self.set_light_off()
-            #         break
                 
-          # turn your sortingrobot list   to the sorted list        
+          # turn your sortingrobot list to the sorted list        
                     
```

Comment:
Great job overall; remember that bubble sort is the worst type of sort we can use here, but it's good for MVP. Try insertion, or better yet, one of the recursive sorts, like merge and quick sort!